### PR TITLE
[MI-3835] Add logic to hide/show RHS on the basis of system console setting

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary

- Added logic to hide/show RHS on the basis of system console settings
- Updated channel header icon
- Wrapped utils scss classes in unique plugin class and updated all the corresponding components and containers
- Fixed unit test cases

### CODE REVIEW

File: webapp/src/index.tsx [hunk 1]
```diff
@@ -30,26 +22,18 @@ export default class Plugin {
...
         registry.registerAdminConsoleCustomSetting('appManifestDownload', MSTeamsAppManifestSetting);
         registry.registerAdminConsoleCustomSetting('ConnectedUsersReportDownload', ListConnectedUsers);
-        const {_, toggleRHSPlugin} = registry.registerRightHandSidebarComponent(Rhs, <RhsTitle/>);
-
-        // TODO: update icons later
-        registry.registerChannelHeaderButtonAction(
-            <img
-                width={24}
-                height={24}
-                src={iconUrl}
```
Comment (SaurabhSharma-884) [inline webapp/src/index.tsx:47]: Add comment here why we are not registering the plugin here

File: webapp/src/App.tsx [hunk 2]
```diff
@@ -43,6 +56,26 @@ const App = (): JSX.Element => {
...
     });
 
+    useApiRequestCompletionState({
+        serviceName: pluginApiServiceConfigs.whitelistUser.apiServiceName,
+        handleSuccess: () => {
+            const {presentInWhitelist} = whitelistUserData as WhitelistUserResponse;
+            if (presentInWhitelist) {
+                const {_, toggleRHSPlugin} = registry.registerRightHandSidebarComponent(Rhs, <RhsTitle/>);
+                registry.registerChannelHeaderButtonAction(
+                    <img
```
Comment (SaurabhSharma-884) [inline webapp/src/App.tsx:66]: Add comments explaining this

File: webapp/src/App.tsx [hunk 3]
```diff
@@ -1,23 +1,34 @@
 import React, {useEffect} from 'react';
+import {Action, Store} from 'redux';
 import {useDispatch} from 'react-redux';
 
-import usePluginApi from 'hooks/usePluginApi';
+// eslint-disable-next-line import/no-unresolved
```
Comment (SaurabhSharma-884) [inline webapp/src/App.tsx:5]: Remove lint rule

Comment (SaurabhSharma-884) [reply webapp/src/App.tsx:5]: - There should be no `export` in global types directory `src/types`
- Remove export of `PluginRegistry`
- Remove import of `PluginRegistry` from any file it should resolve automatically.
- Remove any related lint rules

**Review by SaurabhSharma-884 — CHANGES_REQUESTED** [review]

**Review by SaurabhSharma-884 — APPROVED** [review]

File: webapp/src/App.tsx [hunk 4]
```diff
@@ -1,23 +1,31 @@
...
 
+// global styles
 import 'styles/main.scss';
 
 /**
  * This is the main App component for the plugin
  * @returns {JSX.Element}
  */
-const App = (): JSX.Element => {
+const App = ({registry, store}:{registry: PluginRegistry, store: Store<GlobalState, Action<Record<string, unknown>>>}): JSX.Element => {
```
Comment (avas27JTG) [inline webapp/src/App.tsx:28]: why `unknown`?

File: webapp/src/App.tsx [hunk 5]
```diff
@@ -43,6 +53,28 @@ const App = (): JSX.Element => {
...
+            if (presentInWhitelist) {
+                const {_, toggleRHSPlugin} = registry.registerRightHandSidebarComponent(Rhs, <RhsTitle/>);
+                registry.registerChannelHeaderButtonAction(
+                    <img
+                        width={24}
+                        height={24}
+                        src={iconUrl}
+                        style={{filter: 'grayscale(1)'}}
+                    />, () => store.dispatch(toggleRHSPlugin), null, pluginTitle);
+                if (registry.registerAppBarComponent) {
```
Comment (avas27JTG) [inline webapp/src/App.tsx:71]:
```suggestion

                if (registry.registerAppBarComponent) {
```

**Review by avas27JTG — APPROVED** [review]

File: webapp/src/App.tsx (hunk 4 above)
Comment (ayusht2810) [reply webapp/src/App.tsx:28]: This type was picked from file defining the `Plugin` class, which was taken from the plugin template

File: plugin.json [hunk 6]
```diff
@@ -171,6 +171,12 @@
                       "value": "UserPrincipalName"
                   }
               ]         
+        },{
+            "key": "enableRhs",
+            "display_name": "Enable RHS",
+            "type": "bool",
+            "help_text": "When true, user will be able to view RHS (right hand sidebar).",
```
Comment (avas27JTG) [inline plugin.json:178]:
```suggestion
            "help_text": "When true, users will be able to view RHS (right-hand sidebar).",
```

File: plugin.json [hunk 7]
```diff
@@ -171,6 +171,12 @@
...
+            "default": false
```
Comment (avas27JTG) [inline plugin.json:179]: is this confirmed from Mike that initially don't show to anyone?

File: server/api.go [hunk 8]
```diff
@@ -95,7 +95,7 @@ func NewAPI(p *Plugin, store store.Store) *API {
 	router.HandleFunc("/oauth-redirect", api.oauthRedirectHandler).Methods(http.MethodGet)
 	router.HandleFunc("/connected-users", api.getConnectedUsers).Methods(http.MethodGet)
 	router.HandleFunc("/connected-users/download", api.getConnectedUsersFile).Methods(http.MethodGet)
-	router.HandleFunc("/whitelist-user", api.handleAuthRequired(api.whitelistUser)).Methods(http.MethodGet)
+	router.HandleFunc("/rhs-enabled", api.handleAuthRequired(api.rhsEnabled)).Methods(http.MethodGet)
```
Comment (avas27JTG) [inline server/api.go:98]:
```suggestion
	router.HandleFunc("/config", api.handleAuthRequired(api.getConfig)).Methods(http.MethodGet)
```

this way in future same route can be used to return additional plugin configs.

File: webapp/src/constants/apiService.constant.ts [hunk 9]
```diff
@@ -10,10 +10,10 @@ export const pluginApiServiceConfigs: Record<PluginApiServiceName, PluginApiServ
         method: 'GET',
         apiServiceName: 'needsConnect',
     },
-    whitelistUser: {
-        path: '/whitelist-user',
+    rhsEnabled: {
+        path: '/rhs-enabled',
```
Comment (avas27JTG) [inline webapp/src/constants/apiService.constant.ts:14]:
```suggestion
    getConfig: {
        path: '/rhs-enabled',
```

File: webapp/src/constants/apiService.constant.ts [hunk 10]
```diff
@@ -10,10 +10,10 @@ export const pluginApiServiceConfigs: Record<PluginApiServiceName, PluginApiServ
...
         method: 'GET',
-        apiServiceName: 'whitelistUser',
+        apiServiceName: 'rhsEnabled',
```
Comment (avas27JTG) [inline webapp/src/constants/apiService.constant.ts:16]:
```suggestion
        apiServiceName: 'getConfig',
```

File: plugin.json (hunk 7 above)
Comment (ayusht2810) [reply plugin.json:179]: yes

**Pull request merged**